src/org/zaproxy/zap/spider/Spider.java

```diff
@@ -672,6 +678,14 @@ public boolean isPaused() {
 	 * @return true, if is stopped
 	 */
 	public boolean isStopped() {
+		if (! stopped && this.spiderParam.getMaxDuration() > 0) {
+			// Check to see if the scan has exceeded the specified maxDuration
+			if (TimeUnit.MILLISECONDS.toMinutes(new Date().getTime() - this.timeStarted.getTime()) > 
+					this.spiderParam.getMaxDuration()) {
+				log.info("Spidering process has exceeded maxDuration of " + this.spiderParam.getMaxDuration() + " minute(s)");
+				this.stop();
```

Comment:
It should not call `stop()` directly because `stop()` tries to wait for the worker threads to finish, which are one of the threads calling this method.
It could call `complete()`, if changed to specify if "successful" when notifying the listeners.